src/components/common/navbar.tsx

```diff
@@ -5,142 +5,258 @@ import Image from 'next/image'
 import Menu from '@mui/material/Menu';
 import MenuItem from '@mui/material/MenuItem';
 import logo from '../../../public/navbar_logo.png'
-import Toolbar from '@mui/material/Toolbar';
 import Box from '@mui/material/Box';
 import IconButton from '@mui/material/IconButton';
 import MenuIcon from '@mui/icons-material/Menu';
 import Typography from '@mui/material/Typography';
 import Container from '@mui/material/Container';
 import Button from '@mui/material/Button';
 import MuiAlert, { AlertProps } from '@mui/material/Alert';
+import { useState } from 'react';
 
 const Alert = React.forwardRef<HTMLDivElement, AlertProps>(function Alert(
-  props,
-  ref,
+    props,
+    ref,
 ) {
-  return <MuiAlert elevation={6} ref={ref} variant="filled" {...props} />;
+    return <MuiAlert elevation={6} ref={ref} variant="filled" {...props} />;
 });
 
 const NavBar = () => {
-  const navigatorStyle = "font-serif p-2 cursor-pointer mx-8";
-  const [anchorElNav, setAnchorElNav] = React.useState<null | HTMLElement>(null);
-  
+    const navigatorStyle = "font-serif p-2 cursor-pointer mx-8";
+    const [anchorElNav, setAnchorElNav] = React.useState<null | HTMLElement>(null);
+
     const handleOpenNavMenu = (event: React.MouseEvent<HTMLElement>) => {
-      setAnchorElNav(event.currentTarget);
+        setAnchorElNav(event.currentTarget);
     };
-  
+
     const handleCloseNavMenu = () => {
-      setAnchorElNav(null);
+        setAnchorElNav(null);
+    };
+
+    const [anchorElPaintings, setAnchorElPaintings] = useState<EventTarget & HTMLButtonElement | null>(null);
+    const [anchorElResearch, setAnchorElResearch] = useState<EventTarget & HTMLButtonElement | null>(null);
+    const [anchorElAbout, setAnchorElAbout] = useState<EventTarget & HTMLButtonElement | null>(null);
+
+
+    const handleClickPaintings = (event: React.MouseEvent<HTMLButtonElement>) => {
+        setAnchorElPaintings(() => event.currentTarget);
+    };
+
+    const handleClickResearch = (event: React.MouseEvent<HTMLButtonElement>) => {
+        setAnchorElResearch(() => event.currentTarget);
+    };
+
+    const handleClickAbout = (event: React.MouseEvent<HTMLButtonElement>) => {
+        setAnchorElAbout(() => event.currentTarget);
+    };
```

Comment:
These three methods do the same thing consider combining them into one function and passing in a second param that's either `about`, `paintings` or `research` and in the function definition you could do something like:

Reply:
```
const handleNavBarDropdowns = (
  event: React.MouseEvent<HTMLButtonElement>,
  dropdown_name: string) => {
        if (event === null) {
          setAnchorElAbout(() => event.currentTarget);
          setAnchorElPaintings(() => event.currentTarget);
          setAnchorElResearch(() => event.currentTarget);
        } else if (dropdown_name === 'about') {
          setAnchorElAbout(() => event.currentTarget);
        } else if (dropdown_name === 'paintings') {
          setAnchorElPaintings(() => event.currentTarget);
        } else if (dropdown_name === 'research') {
          setAnchorElResearch(() => event.currentTarget);
        }
};
```

Reply:
Nevermind I don't think this will work anymore